File: iOS-Play-And-Record-Demo-master/iOS-Play-And-Record-Demo/ViewController.m


/* Set this to YES to reproduce the playback issue */
#define IMPEDE_PLAYBACK NO

#import "ViewController.h"
#import <AVFoundation/AVAudioSession.h>
@interface ViewController (){
    float decibel;
  //  AVAudioSession *session;
    
    
}

@property (nonatomic)     AVAudioPlayer       *player;
@property (nonatomic)     AVAudioRecorder     *recorder;
@property (nonatomic)     NSString            *recordedAudioFileName;
@property (nonatomic)     NSURL               *recordedAudioURL;
@property(nonatomic, strong) AVAudioPlayer *backgroundMusic;




@end

@implementation ViewController

@synthesize player, recorder, recordedAudioFileName, recordedAudioURL;

- (void)viewDidLoad
{
    [super viewDidLoad];
    

   
   NSURL *musicFile = [[NSBundle mainBundle] URLForResource:@"music"
                                               withExtension:@"mp3"];
    AVAudioSession *audioSession = [AVAudioSession sharedInstance];
    NSError *setCategoryError = nil;
    [audioSession setCategory:AVAudioSessionCategoryPlayback
                  withOptions:AVAudioSessionCategoryOptionMixWithOthers
                        error:&setCategoryError];
    [audioSession setActive:YES error:nil];

    
//   
//    NSError *err = nil;
//    
//    [session setCategory:AVAudioSessionCategoryPlayAndRecord  withOptions:AVAudioSessionCategoryOptionMixWithOthers|AVAudioSessionCategoryOptionDefaultToSpeaker error:nil];
//    
//    [session setActive:YES error:&err];
   

//    self.backgroundMusic = [[AVAudioPlayer alloc] initWithContentsOfURL:musicFile
//                                                            error:nil];
//    self.backgroundMusic.numberOfLoops = -1;
//    [self.backgroundMusic play];
    
 
    [NSTimer scheduledTimerWithTimeInterval:0.5
                                     target:self
                                   selector:@selector(printDecibel)
                                   userInfo:nil
                                    repeats:YES];
}

#pragma mark Playback


#pragma mark Recording


-(void) printDecibel{
    [recorder updateMeters];
    decibel = [recorder averagePowerForChannel:0];
    _output.text = [NSString stringWithFormat: @"%f", decibel];
    NSLog(@"THE LOG SCORE : %f", decibel);
  
}
- (IBAction)record:(id)sender {
    [self setupAndPrepareToRecord];
    
    [recorder record];
    
    

}

- (void)setupAndPrepareToRecord
{
    if (!IMPEDE_PLAYBACK) {
        [AudioSessionManager setAudioSessionCategory:AVAudioSessionCategoryRecord];
    }

    NSDate *date = [[NSDate alloc] initWithTimeIntervalSinceNow:1];
    recordedAudioFileName = [NSString stringWithFormat:@"%@", date];
    
    // sets the path for audio file
    NSArray *pathComponents = [NSArray arrayWithObjects:
                               [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) lastObject],
                               [NSString stringWithFormat:@"%@.m4a", [self recordedAudioFileName]],
                               nil];
    recordedAudioURL = [NSURL fileURLWithPathComponents:pathComponents];

    // settings for the recorder
    NSMutableDictionary *recordSetting = [[NSMutableDictionary alloc] init];
    [recordSetting setValue:[NSNumber numberWithInt:kAudioFormatMPEG4AAC] forKey:AVFormatIDKey];
    [recordSetting setValue:[NSNumber numberWithFloat:44100.0] forKey:AVSampleRateKey];
    [recordSetting setValue:[NSNumber numberWithInt: 1] forKey:AVNumberOfChannelsKey];
    
    // initiate recorder
    NSError *error;
    recorder = [[AVAudioRecorder alloc] initWithURL:[self recordedAudioURL] settings:recordSetting error:&error];
    [recorder prepareToRecord];
    recorder.meteringEnabled = YES;
}

#pragma mark AVAudioPlayerDelegate

- (void)audioPlayerDidFinishPlaying:(AVAudioPlayer *)player successfully:(BOOL)flag
{
    [[self player] stop];
}

@end

File: waterSensor/waterSensor/ViewController.swift
//
//  ViewController.swift
//  waterSensor
//
//  Created by Ben Ha on 1/14/16.
//  Copyright © 2016 Ben Ha. All rights reserved.
//

import UIKit
import AVFoundation
class ViewController: UIViewController, AVAudioPlayerDelegate,AVAudioRecorderDelegate {
    var audioRecorder:AVAudioRecorder!
    
    let recordSettings = [AVSampleRateKey : NSNumber(float: Float(44100.0)),
        AVFormatIDKey : NSNumber(int: Int32(kAudioFormatMPEG4AAC)),
        AVNumberOfChannelsKey : NSNumber(int: 1),
        AVEncoderAudioQualityKey : NSNumber(int: Int32(AVAudioQuality.Medium.rawValue))]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let audioSession = AVAudioSession.sharedInstance()
        do {
            try audioSession.setCategory(AVAudioSessionCategoryPlayAndRecord)
            try audioRecorder = AVAudioRecorder(URL: self.directoryURL()!,
                settings: recordSettings)
            audioRecorder.prepareToRecord()
        } catch {
        }
    }
    
    func directoryURL() -> NSURL? {
        let fileManager = NSFileManager.defaultManager()
        let urls = fileManager.URLsForDirectory(.DocumentDirectory, inDomains: .UserDomainMask)
        let documentDirectory = urls[0] as NSURL
        let soundURL = documentDirectory.URLByAppendingPathComponent("sound.m4a")
        return soundURL
    }
    func record() {
    
    }
    
    @IBAction func doRecordAction(sender: AnyObject) {
        if !audioRecorder.recording {
            let audioSession = AVAudioSession.sharedInstance()
            do {
                try audioSession.setActive(true)
                audioRecorder.record()
                while(true)
                {
                    print(audioRecorder.averagePowerForChannel(0))
                
                }
            } catch {
            }
        }
    }
    
    @IBAction func doStopAction(sender: AnyObject) {
        audioRecorder.stop()
        let audioSession = AVAudioSession.sharedInstance()
        
        do {
            try audioSession.setActive(false)
        } catch {
        }
    }
}

